bluemira/base/reactor.py

```diff
@@ -20,12 +20,15 @@
 # License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
 """Base class for a Bluemira reactor."""
 
-from typing import Type
+from typing import List, Optional, Type
 from warnings import warn
 
+from rich.progress import track
```

Comment:
can you add rich to our `pyproject.toml` and add a pinned version to our `requirements.txt`. I think its a nice extra dep that is already a dep of a dep anyway